sw/device/lib/runtime/pmp.c

```diff
@@ -0,0 +1,539 @@
+  }
+
+  if (region >= PMP_REGIONS_NUM) {
+    return kPmpRegionConfigureNapotBadRegion;
+  }
+
+  pmp_region_address_t napot_address;
+  pmp_region_configure_napot_result_t napot_result =
+      pmp_napot_address_construct(address, size, &napot_address);
+  if (napot_result != kPmpRegionConfigureNapotOk) {
+    return napot_result;
+  }
+
+  uint32_t field_value = 0;
+  if (!pmp_cfg_permissions_set(config->permissions, &field_value)) {
+    return kPmpRegionConfigureNapotError;
+  }
+
+  pmp_cfg_mode_lock_set(config->lock, &field_value);
+
+  field_value = bitfield_field32_write(field_value, kPmpCfgModeField,
+                                       PMP_CFG_CSR_MODE_NAPOT);
+
+  // Address registers must be written prior to the configuration registers to
+  // ensure that they are not locked.
+  pmp_region_configure_result_t result =
+      pmp_csr_address_write(region, napot_address);
+  if (result != kPmpRegionConfigureOk) {
+    return (pmp_region_configure_napot_result_t)result;
+  }
+
+  result = pmp_csr_cfg_field_write(region, field_value);
+  if (result != kPmpRegionConfigureOk) {
+    return (pmp_region_configure_napot_result_t)result;
+  }
+
+  return kPmpRegionConfigureNapotOk;
+}
+
+pmp_region_configure_result_t pmp_region_configure_tor(
+    pmp_region_index_t region, const pmp_region_config_t *config,
+    pmp_region_address_t address_start, pmp_region_address_t address_end) {
+  if (config == NULL) {
+    return kPmpRegionConfigureBadArg;
+  }
+
+  if (region >= PMP_REGIONS_NUM) {
+    return kPmpRegionConfigureBadRegion;
+  }
+
+  if (region == 0 && address_start > 0) {
+    return kPmpRegionConfigureBadAddress;
+  }
+
+  if (region > 0 && !pmp_address_aligned(address_start)) {
+    return kPmpRegionConfigureBadAddress;
+  }
+
+  if (!pmp_address_aligned(address_end)) {
+    return kPmpRegionConfigureBadAddress;
+  }
+
+  uint32_t field_value = 0;
+  if (!pmp_cfg_permissions_set(config->permissions, &field_value)) {
+    return kPmpRegionConfigureError;
+  }
+
+  pmp_cfg_mode_lock_set(config->lock, &field_value);
+
+  field_value = bitfield_field32_write(field_value, kPmpCfgModeField,
+                                       PMP_CFG_CSR_MODE_TOR);
+
+  // Address registers must be written prior to the configuration registers to
+  // ensure that they are not locked.
+  if (region != 0) {
+    pmp_region_configure_result_t result =
+        pmp_csr_address_write(region - 1, address_start);
+    if (result != kPmpRegionConfigureOk) {
+      return result;
+    }
+  }
+
+  pmp_region_configure_result_t result =
+      pmp_csr_address_write(region, address_end);
+  if (result != kPmpRegionConfigureOk) {
+    return result;
+  }
+
+  result = pmp_csr_cfg_field_write(region, field_value);
+  if (result != kPmpRegionConfigureOk) {
+    return result;
+  }
+
+  return kPmpRegionConfigureOk;
+}
+
+pmp_region_configure_result_t pmp_cfg_mode_lock_status_get(
+    pmp_region_index_t region, pmp_region_lock_t *lock) {
```

Comment:
This does not match the name in the header - it should be `pmp_region_lock_status_get`, I think?

Reply:
Yup, thanks.

Reply:
Done.